sycl/doc/extensions/proposed/sycl_ext_oneapi_improved_host_task.asciidoc

```diff
@@ -0,0 +1,318 @@
+property is used when submitting the host task.
+
+Additionally new member functions are added to the `interop_handle` class for
+adding native events created within the host task. These member functions add
+native events as dependencies to the SYCL `event` returned from the submission
+of the host task as dependent events.
+
+[source,c++]
+----
+template <backend Backend>
+std::vector<backend_return_t<Backend, event>> ext_oneapi_get_native_events();
+----
+
+_Effects_: Returns a `std::vector` of the native events for the backend
+`Backend` for the dependencies for the dependencies to the host task command if
+the host task command was enqueued with the
+`property::host_task::manual_interop_sync` property, otherwise returns an
+empty `std::vector`.
+
+[source,c++]
+----
+template <backend Backend>
+void ext_oneapi_add_native_events(
+  backend_return_t<Backend, event> hostTaskEvent);
+
+template <backend Backend>
+void ext_oneapi_add_native_events(
+  const std::vector<backend_return_t<Backend, event>> &hostTaskEvents)
+----
+
+_Effects_: Adds the native event(s) `hostTaskEvent`/`hostTaskEvents` as an 
+additional dependency to the host task command completion, that will be waited
+on after waiting on the invocation of the host task function any time where the
+host task function would be waited on including `event::wait`, `queue:wait` and
+`queue::wait_and_throw`. If the function is called multiple times all native
+events provided will be waited on collectively.
+
+[NOTE]
+====
+The `std::vector` returned may also be empty if there are no dependencies for
+which there is a native event for the backend.
+====
+
+
+== Example
+
+Below is an example of using the new interfaces using the OpenCL backend.
+
+[source,c++]
+----
+int pattern = 42;
+
+auto e1 = queue.submit([=](sycl::handler &cgh) {
+  accessor acc{bufA, cgh};
+ 
+  cgh.parallel_for<kernelA>([=](sycl::id<1> idx) {
+    acc[0] = 2;
+  });
+});
+
+auto e2 = queue.submit([&](sycl::handler &cgh) {
+  accessor acc{bufB, cgh};
+
+  // creates a dependency on the previous kernel execution
+  cgh.depends_on(e1);
+
+  auto manualInteropSync =
+    ext::oneapi::experimental::property::host_task::manual_interop_sync;
+
+  cgh.host_task([&](sycl::interop_handle &ih, {ext::oneapi::experimental}) {
+    // Dependent events are returned to be synchronized with.
+    auto nativeEvents = ih.get_native_events<backend::opencl>();
+
+    auto queue = ih.get_native_queue<backend::opencl>();
+    auto mem = ih.get_native_mem<backend::opencl>(acc);
+
+    cl_event ne1;
+    clEnqueueFillBuffer(queue, mem, &pattern, sizeof(int), 1 * sizeof(int),
+      1 * sizeof(int), nativeEvents.size(), nativeEvents.data(), &ne1);
+
+    cl_event ne2;
+    clEnqueueReadBuffer(queue, mem, CL_FALSE, 0, sizeof(int), &pattern, 1,
+      &nativeEvent1, &ne2);
+
+    // The event returned by the host task function are waiting on by the event
+    // returned by submit
+    ih.ext_oneapi_add_native_events<backend::opencl>(ne2);
```

Comment:
for a corner case that there are several native events generated and they are independent.
ih.ext_oneapi_add_native_events<backend::opencl>(`nes`);

Is it possible for `e2` to represent all the native events in `nes`?  My understanding is that e2 is just a wrapper of one native event.